component/src/main/java/io/siddhi/extension/store/elasticsearch/ElasticsearchConfigs.java

```diff
@@ -0,0 +1,532 @@
+
+        private ElasticsearchSink elasticsearchSink;
+
+        private BulkProcessorListener(ElasticsearchSink elasticsearchSink) {
+            this.elasticsearchSink = elasticsearchSink;
+        }
+
+        private BulkProcessorListener() {
+
+        }
+
+        @Override
+        public void beforeBulk(long executionId, BulkRequest request) {
+
+            int numberOfActions = request.numberOfActions();
+            if (logger.isDebugEnabled()) {
+                logger.debug("Executing bulk [{" + executionId + "}] with {" + numberOfActions + "} requests");
+            }
+        }
+
+        @Override
+        public void afterBulk(long executionId, BulkRequest request, BulkResponse response) {
+
+            if (response.hasFailures()) {
+                logger.warn("Bulk [{}] executed with failures for executionId: " + executionId + ", failure : "
+                        + response.buildFailureMessage() + ", status : " + response.status().getStatus());
+                if (logger.isDebugEnabled()) {
+                    for (BulkItemResponse itemResponse : response) {
+                        if (itemResponse.isFailed()) {
+                            logger.warn("Bulk [{}] executed with failures for executionId: " + executionId
+                                    + ", item : " + itemResponse.getItemId() + ", response message: "
+                                    + itemResponse.getFailureMessage() + ", failure : " + itemResponse.getFailure()
+                                    + " message : " + itemResponse.getResponse().toString());
+                        }
+                    }
+                }
+            } else {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Bulk [{" + executionId + "}] completed in {" + response.getTook().getMillis() +
+                            "} milliseconds");
+                    for (BulkItemResponse itemResponse : response) {
+                        logger.trace("Bulk [{" + executionId + "}] completed for : " +
+                                itemResponse.getResponse().toString());
+                    }
+                }
+            }
+        }
+
+        @Override
+        public void afterBulk(long executionId, BulkRequest request, Throwable failure) {
+
+            if (request.requests().size() > 0) {
+                if (failure instanceof Exception && request.requests().get(0) instanceof SiddhiIndexRequest) {
+                    for (DocWriteRequest docWriteRequest : request.requests()) {
+                        SiddhiIndexRequest siddhiIndexRequest = (SiddhiIndexRequest) docWriteRequest;
+                        elasticsearchSink.onError(siddhiIndexRequest.getPayload(), siddhiIndexRequest
+                                .getDynamicOptions(), (Exception) failure);
+                    }
+                }
+            }
+            logger.error("Failed to execute bulk", failure);
```

Comment:
Improve the error message.

Reply:
Fixed in https://github.com/siddhi-io/siddhi-store-elasticsearch/pull/41/commits/2c90b5856b203a9ad479c745f00b96442646fa04